py/torch_tensorrt/dynamo/runtime/_PythonTorchTensorRTModule.py

```diff
@@ -260,13 +272,22 @@ def set_default_device_memory_budget(self) -> int:
         return self._set_device_memory_budget(budget_bytes)
 
     def setup_engine(self) -> None:
+
+        if isinstance(self.engine, trt.ICudaEngine):
```

Comment:
We need to think about how this will interact with the greater system like the `lazy_engine_init` option.